crates/notion-core/src/shell/fish.rs

```diff
@@ -0,0 +1,71 @@
+use std::path::{Path, PathBuf};
+
+use super::{Postscript, Shell};
+
+pub(crate) struct Fish {
+    pub(crate) postscript_path: PathBuf,
+}
+
+impl Shell for Fish {
+    fn postscript_path(&self) -> &Path {
+        &self.postscript_path
+    }
+
+    fn compile_postscript(&self, postscript: &Postscript) -> String {
+        match postscript {
+            &Postscript::Activate(ref s) => {
+                let updated_path = format!("set -U fish_user_paths '{}'\n", s);
+                updated_path + "set -U NOTION_HOME \"$HOME/.notion\"\n"
+            }
+            &Postscript::Deactivate(ref s) => {
+                format!("set -U fish_user_paths '{}'\nset -e NOTION_HOME\n", s)
+            }
+            &Postscript::ToolVersion {
+                ref tool,
+                ref version,
+            } => format!(
+                "set -U NOTION_{}_VERSION {}\n",
+                tool.to_ascii_uppercase(),
+                version
+            ),
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use semver::Version;
+    use std::str::FromStr;
+
+    use crate::shell::{CurrentShell, Postscript, Shell};
+
+    #[test]
+    fn test_compile_postscript() {
+        let fish = CurrentShell::from_str("fish").expect("Could not create fish shell");
+
+        assert_eq!(
+            fish.compile_postscript(&Postscript::Deactivate("some:path".to_string())),
+            "set -U fish_user_paths 'some:path'\nset -e NOTION_HOME\n"
+        );
+
+        assert_eq!(
```

Comment:
Just so it doesn't get lost:  the "ISSUE(99)" comment should probably go here, as well.

Reply:
Ah, good point; will do.